### DISCUSSION TITLE
errorFormatter not going through transformer?

### DISCUSSION THREAD

**Discussion by arily** [trpc/trpc#4058] [Q&A]
I am using zod as a validator, and superjson as transformer.

I'm running into a problem which error has not be transformed by superjson, the "shape" of the error are different between wether superjson has been enabled or not.

![image](https://user-images.githubusercontent.com/10007589/227073518-8961a473-2b26-4510-962b-3f9b19525858.png)
without superjson

![image](https://user-images.githubusercontent.com/10007589/227073629-72632794-d4d9-4cb9-a4fe-8661d8c2ebba.png)
with superjson.

my errorFormatter
![image](https://user-images.githubusercontent.com/10007589/227073902-dfe30654-b93f-45a9-8a88-725f76b8e8ff.png)

**Comment by KATT** [comment]
This looks like correct behavior to me?

When enabling superjson, the error looks like it's wrapped with the json prop etc?

Is it not propagating to the client as expected?

**Reply by arily** [reply]
Hi,
Could you please check if the behavior in my reply is working as intended?

**Reply by KATT** [reply]
Done

**Comment by arily** [comment]
Hi,
Thank you for your answer!
I thought the "transformer" would be transparent to server / client (with no error it is by an extent), thus will have the same output with or without superjson. but I have to use `cause.data.error.json` with superjson and `cause.data.error` without it.
also I am not getting the inferred type correctly, is this expected too?

![image](https://user-images.githubusercontent.com/10007589/228705432-8d2fafb2-9939-498e-8ea7-655cc00b0d7f.png)
maybe I did something wrong so I attached my code.

**Reply by KATT** [reply]
It should be transparent if you use superjson on both ends, could you make a reproduction and a bug report?
